core/modules/wdb/QueryRunner.cc

```diff
@@ -322,29 +324,33 @@ void QueryRunner::_transmit(bool last, unsigned int rowCount, size_t tSize) {
     _result->set_attemptcount(_task->getAttemptCount());
 
     if (!_multiError.empty()) {
-        std::string chunkId = std::to_string(_task->msg->chunkid());
-        std::string msg = "Error(s) in result for chunk #" + chunkId + ": " + _multiError.toOneLineString();
+        string chunkId = to_string(_task->msg->chunkid());
+        string msg = "Error(s) in result for chunk #" + chunkId + ": " + _multiError.toOneLineString();
         _result->set_errormsg(msg);
         LOGS(_log, LOG_LVL_ERROR, msg);
     }
     _result->SerializeToString(&resultString);
-    _transmitHeader(resultString);
-    LOGS(_log, LOG_LVL_DEBUG, "_transmit last=" << last
-         << " resultString=" << util::prettyCharList(resultString, 5));
-
-    if (!_cancelled) {
-        // StreamBuffer::create invalidates resultString by using std::move()
-        xrdsvc::StreamBuffer::Ptr streamBuf(xrdsvc::StreamBuffer::createWithMove(resultString));
-        _sendBuf(streamBuf, last, transmitHisto, "body");
-    } else {
-        LOGS(_log, LOG_LVL_DEBUG, "_transmit cancelled");
+    {
+        // Nothing else can use this channel until this transmit is done.
+        lock_guard<mutex> streamLock(task->sendChannel->getStreamMutexRef());
+        _transmitHeader(resultString);
+        LOGS(_log, LOG_LVL_DEBUG, "_transmit last=" << last
```

Comment:
C++ has a built-in function `__func__` that captures the name of the current context:
```
LOGS(_log, LOG_LVL_DEBUG, __func__ << " last=" << last
```
It may help you in the future in two scenarios:
* if you will decide to rename the function `_transmit`) to something else. Then you won't need to chance all occurrences of the function names in the debug messages.
* if you will cut-and-paste the printout message with an intent to make a similar printout from some other method and forget to rename `_transmit` to match the name of a new host method.
The predefined compiler-time constant `__func__` prevents these and many other potential problems.

Reply:
For the most part, including the function name doesn't make that much difference as the log includes file name and line number. However, using __func__ can certainly be useful in many cases where the message context is vague.